packages/network-controller/tests/NetworkController.provider.test.ts

```diff
@@ -0,0 +1,894 @@
+      },
+    );
+  });
+
+  it('does not transition the status of a network client from "degraded" the first time a failover is activated but it does not return a 2xx response', async () => {
+    const primaryEndpointUrl = 'https://first.endpoint';
+    const secondaryEndpointUrl = 'https://second.endpoint';
+    const networkClientId = 'AAAA-AAAA-AAAA-AAAA';
+    const rpcMethod = 'eth_gasPrice';
+
+    nock(primaryEndpointUrl)
+      .post('/', {
+        id: /^\d+$/u,
+        jsonrpc: '2.0',
+        method: 'eth_blockNumber',
+        params: [],
+      })
+      .times(15)
+      .reply(503);
+    nock(secondaryEndpointUrl)
+      .post('/', {
+        id: /^\d+$/u,
+        jsonrpc: '2.0',
+        method: 'eth_blockNumber',
+        params: [],
+      })
+      .times(5)
+      .reply(503);
+
+    await withController(
+      {
+        isRpcFailoverEnabled: true,
+        state: {
+          networkConfigurationsByChainId: {
+            '0x1337': buildCustomNetworkConfiguration({
+              chainId: '0x1337',
+              name: 'Test Network',
+              rpcEndpoints: [
+                buildCustomRpcEndpoint({
+                  networkClientId,
+                  url: primaryEndpointUrl,
+                  failoverUrls: [secondaryEndpointUrl],
+                }),
+              ],
+            }),
+          },
+          networksMetadata: {
+            [networkClientId]: {
+              EIPS: {},
+              status: NetworkStatus.Unknown,
+            },
+          },
+          selectedNetworkClientId: networkClientId,
+        },
+      },
+      async ({ controller, messenger }) => {
+        messenger.subscribe('NetworkController:rpcEndpointRetried', () => {
+          clock.next();
+        });
+        const stateChangeListener = jest.fn();
+        messenger.subscribe(
+          'NetworkController:stateChange',
+          stateChangeListener,
+        );
+        const { provider } = controller.getNetworkClientById(networkClientId);
+        const request = {
+          id: 1,
+          jsonrpc: '2.0' as const,
+          method: rpcMethod,
+          params: [],
+        };
+        const expectedError = 'RPC endpoint not found or unavailable';
+
+        // Hit the primary, run out of retries.
+        await expect(provider.request(request)).rejects.toThrow(expectedError);
+        // Hit the primary, run out of retries.
+        await expect(provider.request(request)).rejects.toThrow(expectedError);
+        // Hit the primary, break the circuit, fail over to the secondary,
+        // run out of retries.
+        await expect(provider.request(request)).rejects.toThrow(expectedError);
+
+        expect(controller.state.networksMetadata[networkClientId].status).toBe(
+          NetworkStatus.Degraded,
+        );
+      },
+    );
+  });
+
+  it('does not transition the status of a network client from "degraded" the first time a failover is activated but requests are slow to complete', async () => {
```

Comment:
Nit: Another attempted simplification:

```suggestion
  it('does not transition the status of a network client from "degraded" the first time a failover is activated if requests are slow to complete', async () => {
```

Reply:
fixed here https://github.com/MetaMask/core/pull/7186/commits/e623211520efd0b84e86125db995a0b3bd3736dd